crates/transaction-pool/src/pool/events.rs

```diff
@@ -1,13 +1,13 @@
 use crate::{traits::PropagateKind, PoolTransaction, ValidPoolTransaction};
-use reth_primitives::{TxHash, B256};
+use reth_primitives::{TransactionSignedEcRecovered, TxHash, B256};
 use std::sync::Arc;
 
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 
 /// An event that happened to a transaction and contains its full body where possible.
 #[derive(Debug)]
-pub enum FullTransactionEvent<T: PoolTransaction> {
+pub enum FullTransactionEvent<T: PoolTransaction<Consensus = TransactionSignedEcRecovered>> {
```

Comment:
similarly here events itself do not rely on `Consensus = TransactionSignedEcRecovered`, only their consumers are, and they can enforce `Consensus` bound on their side